pkg/pillar/cmd/client/client.go

```diff
@@ -516,31 +546,96 @@ func selfRegister(zedcloudCtx zedcloud.ZedCloudContext, tlsConfig *tls.Config, d
 		log.Errorf("%s\n", string(contents))
 		done = false
 	}
+
 	return done
 }
 
-func doGetUUID(zedcloudCtx zedcloud.ZedCloudContext, tlsConfig *tls.Config,
+// fetch V2 certs from cloud, return GotCloudCerts and ServerIsV1 boolean
+// if got certs, the leaf is saved to types.ServerCertFileName file
+func fetchCertChain(zedcloudCtx zedcloud.ZedCloudContext, tlsConfig *tls.Config, retryCount int, force bool) bool {
+	var resp *http.Response
+	var b, contents []byte
+	var done bool
+
+	if !force {
```

Comment:
FWIW all the callers seem to set force

Reply:
Right. For now. But i was thinking if we all move to V2, if the cert-filename exist, we don't fetch first. most likely it won't change. if changes, let doGetUUID to trigger a re-fetch later. thus, we may use false later.